### DISCUSSION TITLE
min_road_class for routing

### DISCUSSION THREAD

**Discussion by nikhil-potterstech** [valhalla/valhalla#4889] [Q&A]
In here, https://valhalla.github.io/valhalla/api/turn-by-turn/api-reference/  
`min_road_class` comes under `search_filter`, coming in a table that appears under Locations section. 

It's giving the impression that we have to include it in the locations array, next to each location.

I also found one occurence in the code that points to the same:
https://github.com/valhalla/valhalla/blob/5a849793e047451febde2994130254deef57072b/test/gurka/test_search_filter.cc#L150

But what I want actually is to tell valhalla to avoid roads falling below the provided min_road_class. This structure doesn't look like it'll do that; rather it seems this is just for finding the starting / ending point on the road network corresponding to the given lat-lon location.

Can anyone who knows more give clarity on this?

Simple problem statement: I want valhalla to give a route that only uses roads tagged `highway=tertiary` or above. What to put in the api request json to make this happen?

**Comment by chrstnbwnkl** [comment]
You're right about the search filters only filtering out location candidates, without any influence on the actual path finding. What you are looking for currently does not exist as a request option in Valhalla. You could build your own graph from a custom OSM file that only contains the roads you want.
